### PR TITLE
Add SOURCE_REF variable to builder container

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: #3176

### CODE REVIEW

**Comment by mfojtik** [comment]
@soltysh @bparees PTAL

**Comment by mfojtik** [comment]
[test]

**Comment by soltysh** [comment]
LGTM, but please fix the tests :wink:

**Comment by mfojtik** [comment]
unit test fixed...

**Comment by soltysh** [comment]
Thank you [merge].

File: pkg/build/controller/strategy/custom_test.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func TestCustomCreateBuildPod(t *testing.T) {
 	errorCases := map[int][]string{
 		0: {"BUILD", string(buildJSON)},
 	}
-	standardEnv := []string{"SOURCE_URI", "SOURCE_REF", "OUTPUT_IMAGE", "OUTPUT_REGISTRY"}
+	standardEnv := []string{"SOURCE_URI", "SOURCE_REPOSITORY", "SOURCE_CONTEXT_DIR", "SOURCE_REF", "OUTPUT_IMAGE", "OUTPUT_REGISTRY"}
```
Comment (bparees) [inline pkg/build/controller/strategy/custom_test.go]: why do we have source_uri and source_repository?  was this a legacy/compatibility thing?

Comment (mfojtik) [inline pkg/build/controller/strategy/custom_test.go]: good question. I can't see that variable defined anywhere... so I dunno how can this test pass.

Comment (soltysh) [inline pkg/build/controller/strategy/custom_test.go]: SOURCE_URI - https://github.com/openshift/origin/blob/master/pkg/build/controller/strategy/util.go#L54
IIRC, initially we set it for users to get the repo information.

Comment (bparees) [inline pkg/build/controller/strategy/custom_test.go]: so it's redundant now, can you add a comment marking it as deprecrated at least?

**Pull request merged**